has C-linkage specified, but returns user-defined type which is incompatible with C

Hi! I have project where I'm trying to expose an 'interface' on the Rust side for C++ to implement. This exposes a C++ header and opaque type which looks something like

```cpp
#pragma once
#include "rust/cxx.h"

namespace my_namespace
{
  class MyInterface {
  public:
    virtual void doSomething(rust::Fn<rust::Vec<uint8_t>(std::vector<uint8_t> const & inputData)> dataProcessor) const = 0;

    virtual ~MyInterface(){};
  };
}
```

Essentially, it takes a function pointer to an operation implemented in Rust. It's exposed in Rust as

```rust
unsafe extern "C++" {
  include!(<myheader.h>);
  type MyInterface;
  #[rust_name="do_something"]
  fn doSomething(&self, data_processor: fn(&CxxVector<u8>) -> Vec<u8>;
}
```

While this code empirically works, when compiling it issues a warning, complaining that:

```
warning: /myproject-172d70075ad9da85/out/cxxbridge/sources/myproject/src/lib.rs.cc:1417:29: warning: 'my_namespace$cxxbridge1$MyInterface$do_something$data_processor$0' has C-linkage specified, but returns user-defined type '::rust::Vec< ::std::uint8_t>' (aka 'Vec<unsigned char>') which is incompatible with C [-Wreturn-type-c-linkage]
warning: ::rust::Vec<::std::uint8_t> mynamespace$cxxbridge1$MyInterface$do_something$data_processor$0( ::std::vector<::std::uint8_t> const &arg0, void *extern$) noexcept {
warning:                             ^
warning: 1 warning generated.
```

Is this related to my declaration or is it a limitation of CXX right now? Is anything bad going to happen? Like I assume many of you, I'm much more familiar with Rust than C++.

--- a/gen/src/pragma.rs
+++ b/gen/src/pragma.rs
@@ -6,6 +6,7 @@ pub(crate) struct Pragma<'a> {
     pub gnu_diagnostic_ignore: BTreeSet<&'a str>,
     pub clang_diagnostic_ignore: BTreeSet<&'a str>,
     pub dollar_in_identifier: bool,
+    pub return_type_c_linkage: bool,
     pub begin: Content<'a>,
     pub end: Content<'a>,
 }
@@ -21,6 +22,12 @@ pub(super) fn write(out: &mut OutFile) {
         out.pragma
             .clang_diagnostic_ignore
             .insert("-Wdollar-in-identifier-extension");
+    }
+
+    if out.pragma.return_type_c_linkage {
+        out.pragma
+            .clang_diagnostic_ignore
+            .insert("-Wreturn-type-c-linkage");
     }
 
     let begin = &mut out.pragma.begin;

--- a/gen/src/write.rs
+++ b/gen/src/write.rs
@@ -956,6 +956,8 @@ fn write_cxx_function_shim<'a>(out: &mut OutFile<'a>, efn: &'a ExternFn) {
 }
 
 fn write_function_pointer_trampoline(out: &mut OutFile, efn: &ExternFn, var: &Pair, f: &Signature) {
+    out.pragma.return_type_c_linkage = true;
+
     let r_trampoline = mangle::r_trampoline(efn, var, out.types);
     let indirect_call = true;
     write_rust_function_decl_impl(out, &r_trampoline, f, indirect_call);

